tests/health_check.rs

```diff
@@ -0,0 +1,47 @@
+use std::net::TcpListener;
+
+use anyhow::Result;
+
+use hilow;
+
+#[actix_rt::test]
+async fn health_check_works() {
+    let address = spawn_app().expect("Unable to spawn server");
+    let client = reqwest::Client::new();
+
+    let response = client
+        .get(&format!("{}/health_check", address))
+        .send()
+        .await
+        .expect("Failed to  execute request.");
+
+    assert!(response.status().is_success());
+}
+
+#[actix_rt::test]
+async fn health_check_works_failure() {
+    std::env::set_var("DATABASE_URL", "postgres://hilow:h@127.0.0.1/hilow");
+    let address = spawn_app().expect("Unable to spawn server");
+
+    let client = reqwest::Client::new();
+
+    let response = client
+        .get(&format!("{}/health_check", address))
+        .send()
+        .await
+        .expect("Failed to  execute request.");
+
+    assert!(response.status().is_server_error());
+}
+
+/// Spawn a hilow server and return its address to the caller.
+fn spawn_app() -> Result<String> {
+    // Port 0 translates to a random port.
+    let listener = TcpListener::bind("127.0.0.1:0").expect("Failed to bind random port");
```

Comment:
Because if we use 7878, we cannot run tests locally while we have the process running. The port `0` will make that the OS looks for a random available port and use that. I don't see how this could make this function to fail.

Reply:
Is there a need to run our tests in parallel? Are you having any trouble running this tests with the previous server implementation?

Reply:
no, but even with --test-threads=1 we can have problems if we use the same port for each test.

Reply:
Regularly each test in Rust runs on its own thread but as we are using Actix Runtime which runs on Tokio, even if you decide to consume 1 thread Tokio is able to spawn N green threads (Tokio tasks) in the same thread.

Actually you shouldn't have to worry about this because Actix provides its own runtime which is used to solve this issues.

Please read on: https://docs.rs/actix-rt/2.1.0/actix_rt/index.html for more details, but we should be spawning this using Tokio ourselves, we just have to mark our test functions with `actix_rt::test` and leave Actix Runtime handle this, we should focus on write our tests and make them pass.

These tests should look as straightforward as:

```rust
    #[actix_rt::test]
    async fn test_index_get() {
        let mut app = test::init_service(App::new().route("/", web::get().to(index))).await;
        let req = test::TestRequest::with_header("content-type", "text/plain").to_request();
        let resp = test::call_service(&mut app, req).await;
        assert!(resp.status().is_success());
    }
```

[Reference](https://actix.rs/docs/testing/)